fastify.js

```diff
@@ -835,15 +860,17 @@ function build (options) {
       this[kLogLevel]
     )
 
-    app.once('preReady', () => {
+    avvio.once('preReady', () => {
       const context = this[kFourOhFourContext]
 
       const onRequest = this[kHooks].onRequest
-      const preHandler = this[kHooks].preHandler.concat(opts.beforeHandler || [])
+      const preValidation = this[kHooks].preValidation.concat(opts.preValidation || [])
+      const preHandler = this[kHooks].preHandler.concat(opts.beforeHandler || opts.preHandler || [])
```

Comment:
The `opts.beforeHandler` hooks get bound to the fastify instance here: https://github.com/fastify/fastify/blob/4e047a873ebf4cd2eb5bcece57425704e62d35cc/fastify.js#L824-L832

However, the `opts.preHandler` hooks aren't bound. It looks like the code above should be made the same as how you [handled](https://github.com/fastify/fastify/pull/1153/files#diff-20da0b6140a4768d182af0d9b185ac28R598) `before/preHandler` hooks for normal routes.